server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/CreateSnapshotRequest.java

```diff
@@ -364,6 +373,23 @@ public CreateSnapshotRequest userMetadata(@Nullable Map<String, Object> userMeta
         return this;
     }
 
+    /**
+     * Set a uuid to identify snapshot.
+     * If no uuid is specified, one will be created within SnapshotService
+     */
+    public CreateSnapshotRequest uuid(String uuid) {
+        this.uuid = uuid;
+        return this;
+    }
+
+    public String uuid() {
+        // If request was constructed on node with transport version < PRE_REGISTER_SLM_STATS it will not have a uuid yet
```

Comment:
Set the uuid in the getter to deal with the rare case that the node that created the request had a transport version before PRE_REGISTER_SLM_STATS and thus didn't set uuid.